src/app.rs

```diff
@@ -246,6 +246,19 @@ impl<State: Send + Sync + 'static> App<State> {
         http_service_hyper::run(self.into_http_service(), addr);
         Ok(())
     }
+
+    /// Asynchronously serve the app at the given address.
+    #[cfg(feature = "hyper")]
+    pub async fn serve(self, addr: impl std::net::ToSocketAddrs) -> std::io::Result<()> {
+        let addr = addr
+            .to_socket_addrs()?
+            .next()
+            .ok_or(std::io::ErrorKind::InvalidInput)?;
+
+        // TODO: propagate the error from hyper
+        await!(http_service_hyper::serve(self.into_http_service(), addr)).ok();
```

Comment:
This will need to use the new await syntax:

```rust
http_service_hyper::serve(self.into_http_service(), addr).await.ok();
```

Reply:
Done!